cmake/machinecfg/CI.cmake

```
@@ -20,7 +20,7 @@ message(STATUS "Loading machine configuration for default CI machine. "
 
 # common options
 set(Kokkos_ARCH_WSM ON CACHE BOOL "CPU architecture")
-set(NUM_MPI_PROC_TESTING ON CACHE STRING "1")
+set(NUM_MPI_PROC_TESTING "1")
```

Make this change and then you shouldn't need to do all the checking above about checking the cache. You just had the arguments to set messed up before - sorry I didn't notice it.

```suggestion
set(NUM_MPI_PROC_TESTING "1" CACHE STRING "CI runs tests with 1 MPI rank")
```